js/src/index.ts

```diff
@@ -6,36 +6,51 @@
 //
 //
 
-import { 
-    Connection, 
+import {
+    Connection,
     Keypair,
-    PublicKey, 
+    PublicKey
 } from '@solana/web3.js';
-import { 
-    NautilusQuery, 
+import { NautilusIdl, NautilusProgram } from './types';
+import {
+    NautilusQuery,
     NautilusTable,
 } from './sql';
-import { NautilusUtils } from './util';
 
-export class Nautilus {
+import { decapitalizeFirstLetter } from './util';
+
+export class Nautilus<Program extends NautilusProgram = NautilusProgram> {
 
     connection: Connection;
-    programs: [PublicKey, string][];
-    defaultProgram: PublicKey | undefined;
-    payer: Keypair | undefined;
+    programId: PublicKey;
+    programs?: {[programName: string]: PublicKey};

```

Comment:
Curious why you chose to do the string inside an array like that?

Reply:
This means that you can pass a dictionary where keys are strings, e.g: `programs: { tokenMetadata: new PublicKey("XXXX") }` would override the tokenMetadata program. I used this instead of an array of array (`[PublicKey, string][]`) to make it explicit what the key is vs what the value is.